### PR TITLE
DSND-2138: Transform FH delta to child annotation api request

### PR DESCRIPTION
**Pull request by sthompsonCH**
## Describe the changes
This PR implements the transformation and mapping of a child delta to a child annotation API PUT request that is sent to the `filing-history-data-api`.

### Related Jira tickets
[DSND-2138](https://companieshouse.atlassian.net/browse/DSND-2138)

### General
- [x] Is the PR as small as it can be?
- [x] Has the Jira ticket been updated with any relevant comments?
- [ ] Is the `README` up to date?
- [x] Has the `POM` been updated for the latest versions of dependencies?
- [x] Has the code been double-checked against `main`?
- [ ] Does the code adhere standards in this repository, including SonarQube checks?

### Testing
- [x] Do the code changes have unit and integration tests with code coverage > 80%?
- [ ] Has the code been tested locally and/or passed the API Karate tests on Tilt?
- [ ] Have mandatory manual test cases been run?
- [ ] Are extra Karate tests required?
- [ ] Are all the test data resources up to date?

### Release preparation
_Where possible, add links to the respective Jira tickets when an item is checked_
- [ ] Have these changes been included in a release ticket?
- [ ] Are changes required to the `docker-chs-development` deployment or test data?
- [ ] Are any changes required to the environment added to `chs-configs`?

## Notes to the tester
_Add any testing hints or instructions here._

[DSND-2138]: https://companieshouse.atlassian.net/browse/DSND-2138?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by ch-code-analysis** [comment]
CI: Security warnings found!

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Review by kinpang-CH — APPROVED** [review]

File: src/main/java/uk/gov/companieshouse/filinghistory/consumer/mapper/InternalFilingHistoryApiMapper.java [hunk 1]
```diff
@@ -1,86 +1,63 @@
...
-        final TransactionKindResult kindResult = arguments.kindResult();
-        final String encodedId = kindResult.encodedId();
-        final String companyNumber = arguments.companyNumber();
-        final JsonNode topLevelNode = arguments.topLevelNode();
+    public InternalFilingHistoryApi mapInternalFilingHistoryApi(InternalFilingHistoryApiMapperArguments args) {
+        JsonNode topLevelNode = args.topLevelNode();
+
+        TransactionKindResult kindResult = Optional.ofNullable(args.kindResult())
+                .orElseGet(() -> {
+                    LOGGER.error("Null transaction kind result provided");
```
Comment (johncook-ch) [inline src/main/java/uk/gov/companieshouse/filinghistory/consumer/mapper/InternalFilingHistoryApiMapper.java:37]: `DataMapHolder.getLogMap()` missing

File: src/main/java/uk/gov/companieshouse/filinghistory/consumer/service/FormTypeConfig.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+@Configuration
+public class FormTypeConfig {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(NAMESPACE);
+
+    @Bean
+    public List<String> formTypeBlacklist(@Value("${associated-filings.blacklist}") final String file) {
+        InputStream stream = getClass().getResourceAsStream("/%s".formatted(file));
+        if (stream == null) {
+            LOGGER.error("Blacklist file not found: [%s]".formatted(file));
```
Comment (johncook-ch) [inline src/main/java/uk/gov/companieshouse/filinghistory/consumer/service/FormTypeConfig.java:25]: DataMapHolder.getLogMap()

**Review by johncook-ch — CHANGES_REQUESTED** [review]

**Review by kinpang-CH — APPROVED** [review]

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Review by johncook-ch — DISMISSED** [review]

**Review by kinpang-CH — APPROVED** [review]

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Review by johncook-ch — APPROVED** [review]

**Review by crowlandsCH — APPROVED** [review]

**Review by cgregoryCH — APPROVED** [review]

**Pull request merged**